Wrappers/Python/cil/optimisation/operators/WaveletOperator.py

```diff
@@ -0,0 +1,261 @@
+            'Channels' will compute the wavelet decomposition on only the channels, independently for every spatial point.
+        bnd_cond: str, default 'symmetric'. More commonly known as the padding or extension method used in discrete convolutions. All options supported by PyWavelets are valid.
+            Most common examples are 'symmetric' (padding by mirroring edge values), 'zero' (padding with zeros), 'periodic' (wrapping values around as in circular convolution).
+            Some padding methods can have unexpected effect on the wavelet coefficients at the edges.
+            See https://pywavelets.readthedocs.io/en/latest/ref/signal-extension-modes.html for more details and all options.
+
+        Attributes
+        ----------
+        moments: integer for 
+            number of vanishing moments. Known for Daubechies, None for others
+
+     '''
+
+    def __init__(self, domain_geometry,
+                 range_geometry=None,
+                 level=None,
+                 wname="haar",
+                 axes=None,
+                 **kwargs):
+
+        # Correlation is different way of defining decomposition axes
+        self.correlation = kwargs.get('correlation', None)
+
+        if axes is None and len(domain_geometry.shape) > 1:
+            if self.correlation in [None, 'All']:
+                axes = None
+            elif self.correlation.lower() in ["space", "spatial"]:
+                axes = [i for i, l in enumerate(
+                    domain_geometry.dimension_labels) if l != 'channel']
+            elif self.correlation.lower() in ["channels", "channel"]:
+                axes = [i for i, l in enumerate(
+                    domain_geometry.dimension_labels) if l == 'channel']
+            else:
+                raise AttributeError(
+                    f"Unknown correlation type: '{self.correlation}'")
+            if axes == []:
+                raise AttributeError(
+                    f"Correlation set to '{self.correlation}' but the data only has '{domain_geometry.dimension_labels}' as possible dimensions")
+        elif axes is not None and self.correlation is not None:
+            warnings.warn(
+                f"Decomposition axes '{axes}' take priority over correlation '{self.correlation}'. Both should not be used.", UserWarning)
+        elif len(domain_geometry.shape) == 1 and self.correlation is not None:
+            warnings.warn(
+                f"Setting correlation '{self.correlation}' is not valid for 1D data.", UserWarning)
+
+        # Convolution boundary condition i.e. padding method
+        self.bnd_cond = kwargs.get('bnd_cond', 'symmetric')
+
+        self.wname = wname
+        self._wavelet = pywt.Wavelet(wname)
+        self.moments = self._wavelet.vanishing_moments_psi
+        self._trueAdj = kwargs.get('true_adjoint', True)
```

Comment:
I'm doing this via a comment instead of a PR to make _my_ life bit easier, but I can also do these changes if needed.
- Without the non-orthogonal wavelets the flag `_trueAdj` is not needed and can be discarded (for now).
- The vanishing `moments` are not actually needed for anything either and can be discarded. I originally thought this would be useful but it's not. And the user can always use the `vanishing_moments_psi` function from pywt to access this information.
- `_wavelet` is the pywt wavelet-object. I am not sure if it needs to be hidden from the user or not. A lot of nice functionality and information (from pywt) is behind it but it is also completely determined by the name of the wavelet used.

Reply:
Many thanks @tommheik I will add it in :)